### PR TITLE
script/build-integration-branch: minor improvements around github api access & tokens

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Try to fix some of the annoyances I have with this script wrt token handling. This should be fully backwards compatible with current workflows but allow one to migrate the token to .netrc that may also be used by other tools.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/script/build-integration-branch [hunk 1]
```diff
@@ -52,9 +58,25 @@ except ImportError:
     assert len(sys.argv) == 2
     branch = label + postfix
 
-
-with open(os.path.expanduser('~/.github_token')) as myfile:
-    token = myfile.readline().strip()
+token = ''
+nrc = netrc.netrc()
```
Comment (adk3798) [inline src/script/build-integration-branch:62]: It seems like this fails if you don't have a `.netrc` file in your home dir
```
bash-5.2$ ./script/build-integration-branch wip-adk-testing
Traceback (most recent call last):
  File "/home/adking/orch-ceph/ceph2/src/./script/build-integration-branch", line 62, in <module>
    nrc = netrc.netrc()
  File "/usr/lib64/python3.10/netrc.py", line 30, in __init__
    with open(file, encoding="utf-8") as fp:
FileNotFoundError: [Errno 2] No such file or directory: '/home/adking/.netrc'
```

**Review by adk3798 — CHANGES_REQUESTED** [review]

**Review by adk3798 — APPROVED** [review]
Worked for me, with no netrc stuff set up
```
bash-5.2$ git cherry-pick 96df5875e3065fd10219a9f6c39443f6b8f8e193
[main 727ad841ba0] script/build-integration-branch: add a better error when github api access fails
 Author: John Mulligan <jmulligan@redhat.com>
 Date: Wed Nov 29 15:44:04 2023 -0500
 1 file changed, 5 insertions(+), 1 deletion(-)
bash-5.2$ 
bash-5.2$ git cherry-pick b29e6fbacf26beb3f617200e032e35ed9e7decdf
[main 0b6323ac1a7] script/build-integration-branch: support fetching github token from netrc
 Author: John Mulligan <jmulligan@redhat.com>
 Date: Wed Nov 29 15:45:15 2023 -0500
 1 file changed, 29 insertions(+), 4 deletions(-)
bash-5.2$ 
bash-5.2$ 
bash-5.2$ 
bash-5.2$ ./script/build-integration-branch wip-adk-testing
--- found 9 issues tagged with wip-adk-testing
--- queried 9 prs
branch wip-adk-testing-2023-11-30-1735
--- creating branch wip-adk-testing-2023-11-30-1735
error: branch 'wip-adk-testing-2023-11-30-1735' not found.
Switched to a new branch 'wip-adk-testing-2023-11-30-1735'
--- pr 51963 --- pulling https://github.com/adk3798/ceph.git branch async-timeout-fixup

...

--- done. these PRs were included:
https://github.com/ceph/ceph/pull/51963 - mgr/cephadm: fixups for asyncio based timeout
https://github.com/ceph/ceph/pull/52033 - mgr/cephadm: mark host is offline in osd rm status
https://github.com/ceph/ceph/pull/52815 - qa/cephadm: mgr-nfs-upgrade, match any migration > 2
https://github.com/ceph/ceph/pull/53840 - mgr/cephadm is not defining haproxy tcp healthchecks for Ganesha
https://github.com/ceph/ceph/pull/54334 - qa/cephadm: extend iscsi teuth test
https://github.com/ceph/ceph/pull/54378 - cephadm: move logrotate configs to jinja2 template
https://github.com/ceph/ceph/pull/54441 - cephadm: break various daemon type classes out to smaller files in cephadmlib
https://github.com/ceph/ceph/pull/54679 - src/cephadm: Added rgw for listing in Daemons
https://github.com/ceph/ceph/pull/54681 - python-common/drive_selection: fix limit with existing devices
--- perhaps you want to: ./run-make-check.sh && git push ci wip-adk-testing-2023-11-30-1735
```

**Review by ljflores — APPROVED** [review]

**Comment by phlogistonjohn** [comment]
Thanks for the 2nd review!!

@adk3798 are you OK to merge this?

**Comment by adk3798** [comment]
> Thanks for the 2nd review!!
> 
> @adk3798 are you OK to merge this?

sure, just need to get the CI passing (results are expired, but I'm assuming the failures were unrelated)

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins test windows

**Pull request merged**